src/drivers/intel/cavs/ipc.c

```diff
@@ -133,12 +156,42 @@ static void ipc_irq_handler(void *arg)
 	}
 }
 
+#if CAVS_VERSION > CAVS_VERSION_1_5
+static struct sof_ipc_cmd_hdr *ipc_cavs_read_msg(void)
+{
+	struct sof_ipc_cmd_hdr *hdr = _ipc->comp_data;
+	uint32_t dr;
+	uint32_t dd;
+	uint8_t *payload;
+
+	dr = ipc_read(IPC_DIPCTDR);
+	dd = ipc_read(IPC_DIPCTDD);
+
+	if (dr & IPC_DIPCTDR_MSG_REG) {
+		hdr->cmd = (dr & IPC_DIPCTDR_GLB_CMD_MASK) << 16;
```

Comment:
Macro also for command?
`#define IPC_DIPCTDR_GLB_CMD(x) (((x) & IPC_DIPCTDR_GLB_CMD_MASK) << IPC_DIPCTDR_GLB_CMD_SHIFT)`

Reply:
Will be fixed in a new PR containing new work.